guis/panel/leak/panel_leak_utilities.py

```diff
@@ -92,3 +93,94 @@ def get_time(time_str):
         # df['PSI/degC'] = df["PRESSURE(PSI)"]/df["ROOM TEMPERATURE(C)"]
 
         return df
+
+
+################################################################################
+# READ INPUT
+# Read raw data from the database into a dataframe, structurally equivalent to that of ReadLeakRateFile, but with timestamps retained
+# TIME(DAYS)    FILLPSIA    RefPSIA PRESSURE(PSI)   BOX TEMPERATURE(C)  ROOM TEMPERATURE(C) Heater%
+################################################################################
+
+def readLeakDb(panel, tag):
+    leak_df = pd.DataFrame()
+    
+    #ensure that panel is type str
+    panel = str(panel)
+    
+    try:
+        con = sqlite3.connect('data/database.db')
+        cursor = con.cursor()
+
+        # acquire straw location
+        cursor.execute("SELECT * FROM straw_location WHERE number='"+str(panel)+"' AND location_type='MN'")
+        straw_location = str(cursor.fetchall()[0][0])
+    
+        # use straw location to acquire procedure
+        cursor.execute("SELECT * FROM procedure WHERE straw_location='"+straw_location+"' AND station='pan8'")
+        procedure = str(cursor.fetchall()[0][0])
+        
+        # use procedure and tag to acquire trial
+        cursor.execute("SELECT * FROM panel_leak_test_details WHERE procedure='"+procedure+"' AND tag='"+tag+"'")
+        trial = str(cursor.fetchall()[0][0])
+        
+        # use trial to acquire pertinent entries from measurement_panel_leak
+        leak_df = pd.read_sql_query("SELECT * FROM measurement_panel_leak WHERE trial='"+trial+"'", con)
+        
+        # rename pertinent columns
+        leak_df.rename(columns={"elapsed_days":"TIME(DAYS)"}, inplace=True)
+        leak_df.rename(columns={"pressure_diff":"PRESSURE(PSI)"}, inplace=True)
+        leak_df.rename(columns={"pressure_ref":"RefPSIA"}, inplace=True)
+        leak_df.rename(columns={"pressure_fill":"FillPSIA"}, inplace=True)
+        leak_df.rename(columns={"temp_box":"BOX TEMPERATURE(C)"}, inplace=True)
+        leak_df.rename(columns={"temp_room":"ROOM TEMPERATURE(C)"}, inplace=True)
+        leak_df.rename(columns={"heater_pct":"Heater%"}, inplace=True)
+    
+        # drop other columns
+        leak_df.drop(columns=['id', 'trial'], axis=1, inplace=True)
+        
+        con.close()
+        
+        return leak_df
+    
+    except:
```

Comment:
Two complaints here:
1. not a fan of putting so much code underneath a `try:` statement. It's like you're not sure where you expect a problem to occur. Maybe there's a specific reason for putting so much in it, but I'm not seeing it. Another sneaker reason not to put too much code underneath a try: different languages might still try to process the lines after the exception and before the except (I don't think python does, so this is for other languages) which could produce bad behavior and at least it will be slow.
2. you should catch specific exceptions. looks like we want a sqlite exception of some kind. It's obviously not critical, but it's just good practice. if you tell me that you googled for 1 minute can couldn't find the right syntax for the exception, that's probably a good enough reason for me. And if you really want to catch ANY error and do something about it, you can do `except Exception as e: print(e)` to learn about it.

In summary: just put the one or two lines where the exception can occur underneath the try: and then catch a specific exception if you know what the problem is going to be.